src/test/scala/org/broadinstitute/dsde/workbench/sam/dataAccess/MockDirectoryDAO.scala

```diff
@@ -292,15 +288,13 @@ class MockDirectoryDAO(val groups: mutable.Map[WorkbenchGroupIdentity, Workbench
   override def loadUserByAzureB2CId(userId: AzureB2CId, samRequestContext: SamRequestContext)
       : IO[Option[SamUser]] = IO.pure(users.values.find(_.azureB2CId.contains(userId)))
 
-  override def setUserAzureB2CId(userId: WorkbenchUserId, b2CId: AzureB2CId, samRequestContext: SamRequestContext): IO[Unit] = IO {
-    for {
-      user <- users.get(userId)
-    } yield {
-      users += user.id -> user.copy(azureB2CId = Option(b2CId))
-    }
-  }
+  override def setUserAzureB2CId(userId: WorkbenchUserId, azureB2CId: AzureB2CId, samRequestContext: SamRequestContext)
+  : IO[Unit] = users.get(userId) match {
+        case Some(user) => IO.pure(users += userId -> user.copy(azureB2CId = Option(azureB2CId)))
+        case None => IO.unit
+      }
 
```

Comment:
todo: find usages of this method and run the tests comparing the old and new implementation

Reply:
Tested an reverted